src/main/scala/sbtcrossproject/Platform.scala

```diff
@@ -0,0 +1,49 @@
+package sbtcrossproject
+
+import scala.language.higherKinds
+import sbt._
+
+/**
+ * Created by grinder on 17.07.16.
+ */
+abstract class Platform(val name: String, val root: String)
+
+object JVM extends Platform("jvm", "jvm/src/main") {
+
+  implicit class JvmTypeExt[P <: JVM.type, T1[T <: P]](project: T1[P])(
+      implicit ev$1: T1[P] => CrossProject[P]) {
```

Comment:
This signature seems more complex than necessary to me. In particular the higher-kinded type `T1`. I think the following should be enough:
```scala
implicit class JvmTypeExt[P <: JVM.type](project: CrossProject[P]) {
```